packages/editor/src/client/atoms/permissionControl.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {ComponentType, PropsWithChildren} from 'react'
+import {AuthContext} from '../authContext'
+import {useTranslation} from 'react-i18next'
+
+interface PermissionControlProps {
+  requiredPermission: string
+  showMessage?: boolean
+}
+
+export function PermissionControl({
+  children,
+  requiredPermission,
+  showMessage
+}: PropsWithChildren<PermissionControlProps>) {
+  const {t} = useTranslation()
+  return (
+    <AuthContext.Consumer>
+      {value => (
+        <>
```

Comment:
You could probably refactor the following lines with additional functions. For now the logic is very unreadable.

Reply:
I agree